contracts/hydro/src/contract.rs

```diff
@@ -1827,7 +1827,27 @@ fn enrich_lockups_with_tranche_infos(
                         return None;
                     }
 
-                    let next_round_voting_allowed = next_round_voting_allowed_res.unwrap();
+                    let mut next_round_voting_allowed = next_round_voting_allowed_res.unwrap();
+
+                    // if the next round voting allowed is greater than the current round,
+                    // meaning the lockup has voted on a proposal in some previous round,
+                    // check whether there is a deployment associated with that proposal
+                    if next_round_voting_allowed > current_round_id {
+                        let deployment = find_deployment_for_voted_lock(
+                            deps,
+                            current_round_id,
+                            *tranche_id,
+                            &converted_addr,
+                            lock.lock_entry.lock_id,
+                        )
+                        .expect("Failed to find deployment for voted lock");
```

Comment:
Even though we shouldn't encounter `Err` here, let's replace the `expect()` call with returning None instead (since this chunk of code is within `filter_map()` call. This way our query wouldn't panic in case of unexpected errors. Basically do the same thing as we do when handling all other `Result<>` in this function.

Reply:
feels like it would be weird to just put none here when really something in the store is broken, no?

Reply:
True, but we are doing it in two other cases (if we can't parse `VOTE_MAP` or `VOTING_ALLOWED_ROUND`) so this would make things consistent. On top of that, we would remove `expect()` call which shouldn't be used in production code because it can panic.